### PR TITLE
use bundlesize to ensure bundle does not exceed certain thresholds

### PR DESCRIPTION
**Pull request by angelo-v**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,8 +17,9 @@
     "lint": "eslint 'src/**/*.ts'",
     "lint-fix": "eslint 'src/**/*.ts' --fix",
     "test": "npm run lint",
-    "prepublishOnly": "npm test && npm run build",
-    "postpublish": "git push origin master --follow-tags"
+    "prepublishOnly": "npm test && npm run build && npm run bundlesize",
```
[the remark this answers is not in the document]
Comment (michielbdejong) [reply package.json:20]: Looking at https://github.com/solid/mashlib/blob/master/package.json#L11 I think `npm run build` does run:
* npm run clean
* npm run build-version 
* npm run build-dev 
* npm run build-dist

I think we do need the first two before publishing an npm version. Are you saying skip the third step?

**Comment by michielbdejong** [comment]
I agree with adding the `bundlesize` command but maybe not with failing the release if it's bigger than 1Mb. What should we do when (in the middle of doing a release) the library turns out to be too big? What is its size now?

**Review by michielbdejong — CHANGES_REQUESTED** [review]
Maybe just add the command but not make it block the release?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelo-v) [reply package.json:20]: At least build-dev could be skipped I guess, but let's tackle this in a separate task

**Comment by angelo-v** [comment]
> Maybe just add the command but not make it block the release?

I think it is actually important that it breaks the build, otherwise it will not be regarded.

If it does, we have to options:

 1. Check what caused it and resolve it as far as possible
 2. Deliberatly choose to increase the threshold

I find a deliberate decision very important and want to prevent making the bundle bigger and bigger without noticing directly.

The bundlesize check is included in github, there you can see the current size and limit:

image

**Comment by angelo-v** [comment]
PS: Since it is checked on travis as well, it will not fail in the middle of a release but already in the PR that causes the bundle to exceed size

**Comment by michielbdejong** [comment]
Still, it will fail in mashlib and not in solid-panes or in the pane in question or in solid-ui. Mashlib is an almost empty shell around solid-panes, we only do mashlib PRs to include the latest solid-panes master. But I agree with your reasoning and I guess it's better than nothing. Let's discuss it in tomorrow's Solid OS meeting (every Wednesday 13:00 CEST in https://meet.jit.si/solid-operating-system). Maybe we can also do other things (lazy loading?) that reduce the size.

**Comment by angelo-v** [comment]
The check is in mashlib, because mashlib is what bundles everything up. We cannot include it in the other repos, because we have no bundles there we could measure. I can analyse the bundle to see which dependencies cause the size and do something about it. We can also see how it works out in practise and relax the check if it disturbs more than it helps.

**Comment by angelo-v** [comment]
@michielbdejong so can we merge this as discussed?

**Review by michielbdejong — APPROVED** [review]
Yes, go ahead and merge, as discussed in the Solid OS meeting.

**Pull request merged**